langchain/tools/azure_cognitive_services/image_analysis.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import annotations
+
+import logging
+from typing import Any, Dict, Optional
+
+from pydantic import root_validator
+
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForToolRun,
+    CallbackManagerForToolRun,
+)
+from langchain.tools.azure_cognitive_services.utils import detect_file_src_type
+from langchain.tools.base import BaseTool
+from langchain.utils import get_from_dict_or_env
+
+logger = logging.getLogger(__name__)
+
+
+class AzureCogsImageAnalysisTool(BaseTool):
+    """Tool that queries the Azure Cognitive Services Image Analysis API.
+
+    In order to set this up, follow instructions at:
+    https://learn.microsoft.com/en-us/azure/cognitive-services/computer-vision/quickstarts-sdk/image-analysis-client-library-40
+    """
+
+    azure_cogs_key: str = ""  #: :meta private:
+    azure_cogs_endpoint: str = ""  #: :meta private:
+    vision_service: Any  #: :meta private:
+    analysis_options: Any  #: :meta private:
+
+    name = "Azure Cognitive Services Image Analysis"
+    description = (
+        "A wrapper around Azure Cognitive Services Image Analysis. "
+        "Useful for when you need to analyze images. "
+        "Input should be a url to an image."
+    )
+
+    @root_validator(pre=True)
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and endpoint exists in environment."""
+        azure_cogs_key = get_from_dict_or_env(
+            values, "azure_cogs_key", "AZURE_COGS_KEY"
+        )
+
+        azure_cogs_endpoint = get_from_dict_or_env(
+            values, "azure_cogs_endpoint", "AZURE_COGS_ENDPOINT"
+        )
+
+        try:
+            import azure.ai.vision as sdk
+
+            values["vision_service"] = sdk.VisionServiceOptions(
+                endpoint=azure_cogs_endpoint, key=azure_cogs_key
+            )
+
+            values["analysis_options"] = sdk.ImageAnalysisOptions()
+            values["analysis_options"].features = (
+                sdk.ImageAnalysisFeature.CAPTION
+                | sdk.ImageAnalysisFeature.OBJECTS
+                | sdk.ImageAnalysisFeature.TAGS
+                | sdk.ImageAnalysisFeature.TEXT
+            )
+        except ImportError:
+            raise ImportError(
+                "azure-ai-vision is not installed. "
+                "Run `pip install azure-ai-vision` to install."
```

Comment:
is azure-ai-vision publicly available yet? see it on pypi but seems to be beta versions and i'm not able to pip install it https://pypi.org/project/azure-ai-vision/

Reply:
Yes, it's publicly available. You can find the installation doc here: https://learn.microsoft.com/en-us/azure/cognitive-services/computer-vision/quickstarts-sdk/image-analysis-client-library-40?pivots=programming-language-python&tabs=visual-studio%2Clinux#analyze-image

Which error did you see when you `pip install azure-ai-vision`? I can install it on my conda env

Reply:
```bash
$ pip install azure-ai-vision
ERROR: Could not find a version that satisfies the requirement azure-ai-vision (from versions: none)
ERROR: No matching distribution found for azure-ai-vision
```

Reply:
on m2 mac, python 3.11

Reply:
seems like it may only support windows and linux atm? https://pypi.org/project/azure-ai-vision/0.11.1b1/#files

Reply:
Yes, you're right. This package does not support macos yet. 
Also in the official doc, only win&linux tabs are there: https://learn.microsoft.com/en-us/azure/cognitive-services/computer-vision/quickstarts-sdk/image-analysis-client-library-40?pivots=programming-language-python&tabs=visual-studio%2Clinux#create-environment-variables

Reply:
could we add a check for this? so that we only try to load the AzureCogsImageAnalysisTool if on windows or linux? something like
```python
import sys

...
class AzureCognitiveServicesToolkit(BaseToolkit):
     """Toolkit for Azure Cognitive Services."""

     def get_tools(self) -> List[BaseTool]:
         """Get the tools in the toolkit."""
         
       tools = [
             AzureCogsFormRecognizerTool(),
             AzureCogsSpeech2TextTool(),
             AzureCogsText2SpeechTool(),
         ]
       # TODO: Remove check once azure-ai-vision supports MacOS.
       if sys.platform.startswith("linux") or sys.platform.startswith("win"):
         tools.append(AzureCogsImageAnalysisTool())
       return tools
```

Reply:
and also add a note to the notebook that explains this as well

Reply:
Thanks! Have updated the codes.